### ISSUE TITLE
Re-enabling processing for body connected to joint causes crash

### ISSUE THREAD

**Issue by ShirenY** [godot-jolt/godot-jolt#913]
Godot 4.2.2
Godot Jolt 0.12.0-stable

AT. 
The crash happens when both body connected by joint is diactived at beginning. It happens once one of the rigidbody goes active.
From debugger, I can see when rebuild the joint, body_b's joint_shape is nullptr. Here's the stacktrace,
<img width="1142" alt="InActiveRigidbodyCrash" src="https://github.com/godot-jolt/godot-jolt/assets/3391696/3f597966-8bcc-4320-af91-61b7bd986bf9">

This is the MRP, I can reliablly reproduce this on windows. The plugin binaries are too big for github, so I removed ios and android.

[JointCrash.zip](https://github.com/user-attachments/files/16116335/JointCrash.zip)

**Comment by ShirenY** [comment]
One potential solution I can think of is to check for shapes before rebuilding the joints. However, I'm not entirely certain about its viability.

**Comment by mihe** [comment]
Thank you for reporting this, and for providing such a succinct MRP.

It seems you can repro this even without the script, by simply reverting the `process_mode` property in the editor on any of the two bodies.

I'll have it fixed shortly.

**Comment by mihe** [comment]
I've fixed this in #914, which I'll merge shortly.

It seems that this crash is also reproducible if you simply disable the `node_b` body and restart the editor, which is pretty bad.

The only temporary workaround I can think of, until I can get another release out, is to clear both `node_a` and `node_b` on the joint before you disable the processing of either body, and then restore the node paths before you re-enable the processing, meaning something like this:

```gdscript
func _ready() -> void:
    await get_tree().create_timer(1.0).timeout
    $Joint.node_a = NodePath()
    $Joint.node_b = NodePath()
    $Body2.process_mode = Node.PROCESS_MODE_DISABLED
    await get_tree().create_timer(1.0).timeout
    $Joint.node_a = $Body1.get_path()
    $Joint.node_b = $Body2.get_path()
    $Body2.process_mode = Node.PROCESS_MODE_INHERIT
```

That should produce the same end result, as far as I know.

Also, note that there won't be any more releases targeting Godot 4.2, so you'll need to upgrade to Godot 4.3 to make use of #914 (once I get another release out).

**Comment by ShirenY** [comment]
Thank you for the work. No worries, I already compiled a similiar fix on 4.2 version for myself.
If I may ask, what make godot-jolt moves to 4.3 so early when godot 4.3 is still in development? If I get it correctly, godot 4.3 can still use extensions targets godot 4.2

**Comment by mihe** [comment]
The reason for only targeting a single minor version of Godot, rather than using the default forward-compatibility of GDExtension, is simply because I don't want to deal with keeping track of any potential compatibility breakages, whether that's interface breakages or behavioral breakages, both of which have happened in the past.

The reason for moving over to 4.3 so "early" is exactly because I only target a single minor version of Godot, and people who like to use bleeding-edge versions of Godot keep asking me to upgrade, so I decided to move over the repository itself to 4.3 once 4.3-beta1 was released (and start publishing pre-releases here on GitHub) as that typically indicates that 4.3-stable is only a few weeks away. It was either that or maintain multiple branches, which I don't have the bandwidth for at the moment.

But yes, I'm sure there are people who will be using Godot 4.2 for a while longer, just like there are people who are still using Godot 4.1, and even Godot 3.x. The long-term plan is for this extension to be moved into Godot itself as a first-party module, at which point all this becomes moot anyway.
